### PR TITLE
chore(TUI-149): update jest to 23

### PR DESCRIPTION
**Pull request by jmfrancois**
**What is the problem this PR is trying to solve?**

we are in jest 20, so we can t use new api like it.each

**What is the chosen solution to this problem?**

update to jest 23 in 'versions' script
use version scripts on our dependencies
update some tests to work with jest 23
update all snapshots (this is what makes the number of files so big)
fix some of the failing tests

it speeds up the CI which now take only 10 minutes !!
But it's not only the update its because we use jest in series instead of parallel using `--runInBand`
Sometimes people believe // execution is faster, this is a proof that it's not always the case.

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: npm-audit/package.json [hunk 1]
```diff
@@ -82,8 +82,8 @@
     "eslint-plugin-import": "^1.16.0",
     "eslint-plugin-jsx-a11y": "^2.2.2",
     "eslint-plugin-react": "^6.3.0",
-    "jest": "20.0.3",
-    "jest-cli": "20.0.3",
+    "jest": "^23.0.0",
+    "jest-cli": "^23.0.0",
```
Comment (acateland) [inline npm-audit/package.json]: is the `^` intended ?

Comment (jmfrancois) [reply npm-audit/package.json]: Yes I trust the semver on jest.

**Comment by jmfrancois** [comment]
Now I have time to work on it

File: npm-audit/package.json (hunk 1 above)
Comment (jsomsanith) [reply npm-audit/package.json]: Yeah I think we can trust facebook to follow semver

File: packages/html-webpack-plugin/package.json [hunk 2]
```diff
@@ -17,7 +17,7 @@
     "eslint": "3.4.0",
     "eslint-config-airbnb": "10.0.1",
     "eslint-plugin-import": "1.14.0",
-    "jest-cli": "20.0.3"
+    "jest": "20.0.3"
```
Comment (jsomsanith) [inline packages/html-webpack-plugin/package.json]: The version is ko

File: packages/logging/package.json [hunk 3]
```diff
@@ -56,7 +56,6 @@
     "eslint-plugin-jsx-a11y": "^2.2.2",
     "eslint-plugin-react": "^6.3.0",
     "jest": "20.0.3",
```
Comment (jsomsanith) [inline packages/logging/package.json]: Version is ko

File: packages/storybook-cmf/package.json [hunk 4]
```diff
@@ -40,7 +40,7 @@
     "eslint-plugin-react": "^6.3.0",
     "immutable": "3.8.1",
     "invariant": "2.2.2",
-    "jest": "20.0.3",
+    "jest": "^23.0.0",
```
Comment (jsomsanith) [inline packages/storybook-cmf/package.json]: Why do we have sometimes `"^23.0.0"` and othee times `"^23.6.0"` ?
We should align them

File: package.json [hunk 5]
```diff
@@ -20,7 +20,7 @@
     "build-forms": "lerna exec --scope=@talend/react-forms -- yarn run prepublish",
     "build-icons": "lerna exec --scope=@talend/icons -- yarn run prepublish && lerna exec --scope=@talend/react-components -- yarn test -- -u",
     "build-theme": "lerna exec --scope=@talend/bootstrap-theme -- yarn run prepublish",
-    "test": "lerna exec --concurrency 1 -- TZ=Europe/Paris yarn test",
+    "test": "lerna exec --concurrency 1 -- TZ=Europe/Paris yarn test --detectOpenHandles --forceExit",
```
Comment (jsomsanith) [inline package.json]: I don't really understand this option. Can you explain why you added it ?

Comment (jmfrancois) [reply package.json]: it was to debug the CI which hang with jest. I have finally used --runInBand which fix the issue.
I have used --silent also which speed up the global execution which is now under 10 minutes :D

File: packages/containers/src/ShortcutManager/ShortcutManager.test.js [hunk 6]
```diff
@@ -9,8 +9,9 @@ import Connected from './ShortcutManager.connect';
 
 describe('Container ShortcutManager', () => {
 	it('should render', () => {
-		const wrapper = shallow(<Container />);
+		const wrapper = shallow(<Container redirectMap={{}} />);
 		expect(wrapper.getElement()).toBeNull();
+		wrapper.unmount();
```
[the remark this answers is not in the document]
Comment (jmfrancois) [reply packages/containers/src/ShortcutManager/ShortcutManager.test.js]: to remove the addevent listener which is bind by the didMount. 
I don t know why we don t have the issue in the previous version

File: packages/containers/package.json [hunk 7]
```diff
@@ -72,8 +72,8 @@
     "i18next-parser": "^0.13.0",
     "immutable": "^3.8.1",
     "invariant": "^2.2.2",
-    "jest": "20.0.3",
-    "jest-cli": "20.0.3",
+    "jest": "^23.6.0",
+    "jest-environment-jsdom": "^23.0.0",
```
[the remark this answers is not in the document]
Comment (jmfrancois) [reply packages/containers/package.json]: It s not anymore by default and we have some event listener. So we need it in order to make the test run

File: packages/logging/package.json (hunk 3 above)
Comment (jmfrancois) [reply packages/logging/package.json]: I have added an entry in the travis matrix to test logging and html plugin.

**Pull request merged**